### DISCUSSION TITLE
id3tool vs ffprobe

### DISCUSSION THREAD

**Discussion by szemlicka** [advplyr/audiobookshelf#582] [General]
Hi,

make it sense to use id3tool instead of ffprobe to read the id3tag? I found some performance cheats (below) via google and it looks like this can make some speed improvements.

```
Source https://askubuntu.com/questions/226773/how-to-read-mp3-tags-in-shell
Results (real time):

 Program  | Version    | Time / s
----------+------------+-----------
 exiftool | 10.25      | 49.5 ± 0.5
 lltag    | 0.14.5     | 41   ± 1.0
 ffprobe  | 3.1.3-1+b3 | 33   ± 0.5
 eyeD3    | 0.6.18     | 24   ± 0.5
 id3info  | 3.8.3      | 4.2  ± 0.1
 id3v2    | 0.1.12     | 2.9  ± 0.1
 id3tool  | 1.2a       | 1.7  ± 0.1
 mp3info  | 0.8.5a     | 1.4  ± 0.1

Time-wise the winner here is id3tool (mp3info is faster, but doesn't work with ID3 v2). 
id3v2 is also quite fast, but the getTitleId3 function would need adjustment to also work with ID3v1-tags, which 
may at worst slow it down by factor 2.
```

**Comment by szemlicka** [comment]
No comment here?

**Reply by advplyr** [reply]
id3tool is not packaged for nodejs so someone would have to do that. It also looks like it is only for mp3's and we also need to support mp4 since that is what m4b/m4a audio files are.
Ffprobe will also probe every other audio format like flac, opus, aac, ogg, wma, etc. and give us all the info like audio streams, codecs, chapters, and embedded cover art.

**Reply by szemlicka** [reply]
I understand this. But if the library contains only mp3 and we are talking about tagging performance, id3tool is 10 times faster.  I don't understand so much from programming and can't help here.  Thanks for answering ....:-)
